src/Http/Paginators/OffsetRequestPaginator.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Saloon\Http\Paginators;
+
+use Closure;
+use GuzzleHttp\Promise\PromiseInterface;
+use Saloon\Contracts\Connector;
+use Saloon\Contracts\Request;
+use Saloon\Contracts\Response;
+
+// TODO 1: Look into serialising the Connector and original Request,
+//           to ensure that we can rebuild the paginator state without storing the entire multiverse.
+// TODO 2: Make it easier to extend the RequestPaginator. Preferably via callbacks, so we ideally don't even need separate classes.
```

Comment:
That makes perfect sense. I like the idea and significantly reduces the complexity and issues I had trying to figure out a decent way to handle it, without causing poor DX. 🙂

Reply:
I have implemented it this way, with a base `RequestPaginator`.

I also took the liberty to implement a base `PageRequestPaginator` and `OffsetRequestPaginator`.
Each of which extends the `RequestPaginator`, but implements some of the resolving methods, based on common names (offset, limit, page, etc).

Still WIP, with plenty of TODO to figure out, though.

Reply:
I also think we should make sure we can let people customise the query parameters that are set, nothing fancy maybe properties or a method they extend, but I once dealt with an API that used the terminology "top" and "skip" instead of "limit" and "offset" 🤮